Add Transaction details view

luizreis opened the pull request:
Fixes #98 

#### Changes proposed in this Pull Request

* Add link to Transaction details page in Transactions list view
* Create Transaction details page with card placeholders

#### Testing instructions

* Navigate to Payments > Transactions: to the right of each row you should see an info button
* Clicking on the button should lead you to a page with placeholder cards with transaction details

-------------------

- [x] Tested on mobile (or does not apply)

luizreis committed:
Add payment details page

Due to the way routing works in wc-admin, we have to register a static
transaction details page and pass the transaction id as a query string
parameter, instead of having the id dynamically as part of the url.

luizreis committed:
Add placeholder cards for payment details screen

luizreis committed:
Add link from Transactions view to Payment Details

luizreis committed:
Fetch transaction using getTransactions in Details page

Use getTransactions and filter the result to get the requested transaction
for the Transaction Details page. This is not a good way to fetch a single
transaction and only exists to speed up the development of the page layout.

luizreis committed:
Add OrderLink component

luizreis committed:
Use OrderLink in TransactionList component

luizreis committed:
Fix leading whitespaces

allendav reviewed (changes requested):
Cool. Tests great. A few things to attend to and then this can go in. I also opened #215 to be fixed separately.

luizreis committed:
Create empty transaction object to display details

Remove hacky getTransactions + filter call and pass an empty transaction object to the TransactionDetails component.

The single transaction fetch and use in details component is going to be addressed by issue woocommerce-payments#228.

luizreis committed:
Add no translation required TODO comment to placeholder cards

luizreis committed:
Add link generator TODO commit

luizreis committed:
Move Transaction List inline styles to component stylesheet

luizreis committed:
Change snake_case variables to camelCase

luizreis committed:
Use addQueryArgs to generate transaction details link

allendav reviewed (approved):
:shipit:  after rebasing to resolve conflict
